lib/inspect_task.rb

```diff
@@ -16,11 +16,12 @@
 # you may find current contact information at www.suse.com
 
 class InspectTask
-  def inspect_system(store, host, name, current_user, scopes, options = {})
+  def inspect_system(store, host, name, current_user, scopes, filter_definition, options = {})
```

Comment:
Wouldn't it make more sense to pass a filter object here instead of a filter definition? Then the preparation of the filter could happen in the code calling the `InspectTask` and there would be no need to pass the `:skip_files` option.